tests/conftest.py

```diff
@@ -5682,3 +5719,20 @@ def teardown():
     workload.deploy_workload()
 
     return workload
+
+
+def thread_finalizer(operation, instances):
```

Comment:
not all places that call this function use it as a finalizer. 
I can see that for instance, [here](https://github.com/red-hat-storage/ocs-ci/pull/6159/files#diff-e52e4ddd58b7ef887ab03c04116e676f6280b824ab7469d5d3080e5cba4f2128R845), you are calling it from a non teardown fixture function. 
Maybe woth representing that fact by changing the function name from  `thread_finalizer` to `thread_operation`

Reply:
Sounds good

Reply:
Changed